pkg/cloud/azure/actuators/machine/actuator_test.go

```diff
@@ -470,6 +472,44 @@ func TestAvailabilityZones(t *testing.T) {
 	}
 }
 
+func TestCustomUserData(t *testing.T) {
+	fakeScope := newFakeScope(t, v1alpha1.Node)
+	userDataSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "testCustomUserData",
+			Namespace: "dummyNamespace",
+		},
+		Data: map[string][]byte{
+			"userData": []byte("test-userdata"),
+		},
+	}
+	fakeScope.CoreClient = controllerfake.NewFakeClient(userDataSecret)
+	fakeScope.MachineConfig.UserDataSecret = &corev1.SecretReference{Name: "testCustomUserData"}
+	fakeReconciler := newFakeReconcilerWithScope(t, fakeScope)
+	fakeReconciler.virtualMachinesSvc = &FakeVMCheckZonesService{}
+	if err := fakeReconciler.Create(context.Background()); err != nil {
+		t.Errorf("expected create to succeed %v", err)
```

Comment:
Is this actually `Fatal`? Can the remainder of the tests run gracefully if this fails?

Reply:
Yes it does run gracefully, but it would fail (depending on the tests), in this case it would report list of failures instead of failing at the first test itself.